synapseclient/models/project.py

```diff
@@ -146,23 +153,51 @@ class Project(AccessControllable, StorableContainer):
                 List[datetime],
             ],
         ]
-    ] = None
+    ] = field(default=None, compare=False)
     """Additional metadata associated with the folder. The key is the name of your
     desired annotations. The value is an object containing a list of values
     (use empty list to represent no values for key) and the value type associated with
     all values in the list. To remove all annotations set this to an empty dict `{}`."""
 
+    create_or_update: bool = field(default=True, repr=False)
+    """
+    (Store only)
+
+    Indicates whether the method should automatically perform an update if the resource
+    conflicts with an existing Synapse object. When True this means that any changes
+    to the resource will be non-destructive.
+
+    This boolean is ignored if you've already stored or retrieved the resource from
+    Synapse for this instance at least once. Any changes to the resource will be
+    destructive in this case. For example if you want to delete the content for a field
+    you will need to call `.get()` and then modify the field.
+    """
+
+    parent_id: Optional[str] = None
+    """The parent ID of the project. In practice projects do not have a parent, but this
+    is required for the inner workings of Synapse."""
+
     _last_persistent_instance: Optional["Project"] = field(
         default=None, repr=False, compare=False
     )
     """The last persistent instance of this object. This is used to determine if the
     object has been changed and needs to be updated in Synapse."""
 
+    @property
+    def has_changed(self) -> bool:
+        """Determines if the object has been changed and needs to be updated in Synapse."""
+        return (
+            not self._last_persistent_instance or self._last_persistent_instance != self
+        )
+
     def _set_last_persistent_instance(self) -> None:
         """Stash the last time this object interacted with Synapse. This is used to
         determine if the object has been changed and needs to be updated in Synapse."""
         del self._last_persistent_instance
         self._last_persistent_instance = replace(self)
+        self._last_persistent_instance.annotations = (
+            deepcopy(self.annotations) if self.annotations else None
+        )
```

Comment:
These also are in the folder model, is there a way to have them as part of a mixin or another way to not have the code repeated?

Reply:
Yes - But I am interested to get other models in place before I commit to a mixin for this. In particular table is likely to work the same/similar to this, but I wanted to get to it before making this change.

File is also similar to this - But slightly different in that it also have an `Activity` model which also needs the `dataclasses.replace` function to be called on it.

Reply:
Makes sense to me!